provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -312,11 +312,31 @@ def _get_iface(self, machine_id: str) -> str:
             return 'dummy'
         return ifaces[0]
 
-    # cortx>motr>interface_type
     def _get_iface_type(self, machine_id: str) -> Optional[Protocol]:
-        iface = self.provider.get(
-            'cortx>motr>interface_type',
+        endpoints = self.provider.get(
+            'cortx>hare>hax>endpoints',
             allow_null=True)
+
+        if endpoints is None:
+            return None
+
+        hostname = self.utils.get_hostname(machine_id)
+
+        iface = None
+        # Expected format '<protocol>://<hostname>:<port>'
+        # e.g. endpoints:
+        #      - tcp://data1-node1:22001  # For motr and Hax communication
+        #      - tcp://data1-node2:22001  # For motr and Hax communication
+        for e in endpoints:
+            key = e.split(':')
+
+            if key[0] == 'https':
+                continue
+
+            if key[1].split('/')[2] == hostname:
```

Comment:
Just a suggestion: can we check directly ?
```suggestion
            if hostname in key[1]:
	         iface = key[0]
```

Reply:
`if key[1].split('/')[2] == hostname:`
@Shreya-18 Thanks for the input. Even though original solution looks complex as compare to solution suggested it checks for exact format.
@mssawant Any suggestion here?

Reply:
Looks fine.